### PR TITLE
feature/setup-state-persistence

### PR DESCRIPTION
**Pull request by imranariffin**
* load state from storage on app startup
* save state to storage on every api success action
* improve: save state only when needed

### CODE REVIEW

**Comment by cglotr** [comment]
@imranariffin I think we should go over the app code base during this week's meeting. I need to get ramped up on the code and hopefully, I am able to contribute to client-side this weekend.

**Comment by imranariffin** [comment]
Sounds like a plan. I also would like to give an on overview of it and want
some feedback

File: app/state/reducers/profile/__tests__/code-request.test.js [hunk 1]
```diff
@@ -109,4 +109,37 @@ describe('`profile.codeRequest` reducer', () => {
...
+      const nextState = profileCodeRequestReducer(state, action)
+
+      expect(nextState).toEqual({
+        isLoaded: 'isLoaded',
+        isLoading: 'isLoading',
+        isSuccess: 'isSuccess'
+      })
+    })
+  })
 })
```
Comment (cglotr) [inline app/state/reducers/profile/__tests__/code-request.test.js]: should add end of line

**Comment by imranariffin** [comment]
#24

**Pull request merged**